core/src/scene/styleContext.cpp

```diff
@@ -330,37 +424,18 @@ bool StyleContext::parseStyleResult(StyleParamKey _key, StyleParam::Value& _val)
     } else if (duk_is_null_or_undefined(m_ctx, -1)) {
         // Ignore setting value
         LOGD("duk evaluates JS method to null or undefined.\n");
+    } else if (duk_is_function(m_ctx, -1)) {
+        if (duk_pcall(m_ctx, 0) != 0) {
+            LOGE("EvalFilterFn: %s", duk_safe_to_string(m_ctx, -1));
+            duk_pop(m_ctx);
+        } else {
+            parseStyleResult(_key, _val);
+        }
```

Comment:
This case probably shouldn't be allowed. This means that if a parameter is a function that returns *another* function, that returned function will be evaluated for another result. I don't believe this is how tangram-js would handle this situation and I don't see a reason for us to add it.